### PR TITLE
Add (Enum, str) base class support

### PR DESCRIPTION
**Pull request by jayvdb**

### CODE REVIEW

File: pyrs/transpiler.py [hunk 1]
```diff
@@ -333,6 +333,10 @@ def visit_IntEnum(self, node):
         fields = "\n".join(fields)
         return f"enum {node.name} {{\n{fields}\n}}\n\n"
 
+    def visit_StrEnum(self, node):
+        self._usings.add("strum_macros::EnumString")
+        return f"#[derive(Debug, PartialEq, EnumString)]\n{self.visit_IntEnum(node)}"
```
Comment (adsharma) [inline pyrs/transpiler.py:338]: Rename logic to `visit_Enum` or `visit_CommonEnum` and then have Int/StrEnum reuse the same code.
Also a test case similar to int_enum.py for code coverage in the same commit.

**Review by adsharma — CHANGES_REQUESTED** [review]
Typo in the description: Emum -> Enum

**Comment by adsharma** [comment]
Merged thanks!

**Comment by jayvdb** [comment]
Actually you merged your own version of this, and the generated code failed to compile.

**Pull request closed without merging**